music21/romanText/tsvConverter.py

```diff
@@ -0,0 +1,670 @@
+
+            try:
+                p.measure(measure).insert(offsetInMeasure, thisM21Chord)
+            except:
+                pass
+
+        self.m21stream = s
+
+        return s
+
+    def prepStream(self):
+        '''
+        Prepares a music21 stream for the harmonic analysis to go into.
+        Specifically: creates the score, part, and measure streams,
+        as well as some (the available) metadata based on the original TSV data.
+        '''
+
+        s = stream.Score()
+        p = stream.Part()
+
+        s.insert(0, metadata.Metadata())
+
+        scl0 = self.chordList[0]
+        s.metadata.opusNumber = scl0.op
+        s.metadata.number = scl0.no
+        s.metadata.movementNumber = scl0.mov
+        s.metadata.title = 'Op'+scl0.op+'_No'+scl0.no+'_Mov'+scl0.mov
+
+        startingKeySig = str(self.chordList[0].global_key)
+        ks = key.Key(startingKeySig)
+        p.insert(0, ks)
+
+        currentTimeSig = str(self.chordList[0].timesig)
+        ts = meter.TimeSignature(currentTimeSig)
+        p.insert(0, ts)
+
+        currentMeasureLength = ts.barDuration.quarterLength
+
+        previousMeasure = self.chordList[0].measure - 1  # Covers anacruses
+        for entry in self.chordList:
+            if entry.measure == previousMeasure:
+                continue
+            elif entry.measure != previousMeasure + 1:  # Not every measure has a chord change.
+                for mNo in range(previousMeasure + 1, entry.measure):
+                    m = stream.Measure(number=mNo)
+                    m.offset = currentOffset + currentMeasureLength
+                    p.insert(m)
+
+                    currentOffset = m.offset
+                    previousMeasure = mNo
+            else:  # entry.measure = previousMeasure + 1
+                m = stream.Measure(number=entry.measure)
+                m.offset = entry.totbeat
+                p.insert(m)
+                if entry.timesig != currentTimeSig:
+                    newTS = meter.TimeSignature(entry.timesig)
+                    m.insert(entry.beat - 1, newTS)
+
+                    currentTimeSig = entry.timesig
+                    currentMeasureLength = newTS.barDuration.quarterLength
+
+                previousMeasure = entry.measure
+                currentOffset = entry.totbeat
+
+        s.append(p)
+
+        self.prepdStream = s
+
+        return s
+
+# ------------------------------------------------------------------------------
+
+class M21toTSV:
+    '''
+    Exports a music21 stream to tabular data and (optionally) writes the file.
+    '''
+
+    def __init__(self, m21Stream):
+        self.m21Stream = m21Stream
+        self.tsvData = self.m21ToTsv()
+
+    def m21ToTsv(self):
+        '''
+        Converts a list of music21 chords to a list of lists
+        which can then be written to a tsv file with toTsv(), or processed another way.
+        '''
+
+        tsvData = []
+
+        for thisEntry in self.m21Stream.recurse():
```

Comment:
this should be `.recurse().getElementsByClass('RomanNumeral')` and the inner if statement can be removed.